dev/source/docs/setting-up-sitl-on-macosx.rst

```diff
@@ -0,0 +1,137 @@
+.. _setting-up-sitl-on-macosx:
+
+===========================
+Setting up SITL on Mac OS X
+===========================
+
+This page describes how to setup the :ref:`SITL (Software In The Loop) <sitl-simulator-software-in-the-loop>` on Mac OS X. The specific commands were tested on macOS Catalina 10.15.7 and macOS Big Sur 11.1.
+
+Overview
+========
+
+The SITL simulator allows you to run Plane, Copter or Rover without any
+hardware. It is a build of the autopilot code using an ordinary C++
+compiler, giving you a native executable that allows you to test the
+behaviour of the code without hardware.
+
+SITL runs natively on Linux, Windows and Mac OSX.
+
+.. image:: ../images/SITL_macosx.png
+    :target: ../_images/SITL_macosx.png
+
+
+Install steps
+=============
+
+This guide assumes that a build environment is already setup for Ardupilot. If that is not the case please follow the available guides to setup the build environment for Mac OSX. The following step provide a condensed versions of the steps to take.
+
+
+Prerequisites
+-------------
+
+The following tools need to be installed to complete the installation. Between brackets the version used by the author at time of writing.
+
+ * Xcode (12.4)
+ * Xcode commandline tools (12.4)
+ * Homebrew (3.0.4-52-gbc37074)
+
+Preparing Python
+----------------
+
+Install the required python 3 version using homebrew and activate it. If you are familiar with pyenv or virtualenv it might be better to use one of those to setup the environment. This limits potential problems with other tools that require specific versions of python. How to do that is beyond the scope of this document.
+
+The following steps install python version 3.9.x for system wide usage.
+
+::
+
+	brew install python@3.9
+  export PATH="$(brew --prefix)/opt/python/libexec/bin":$PATH
+	
+	pip install wheel
```

Comment:
This command may require sudo. Best practice to install for the local user is to use the --user option to install for the user only.

Reply:
Not sure how much we want to dive into this in the instructions, but in my experience the best is to isolate ArduPilot-related Python dependencies in a virtualenv; what I do is as follows:

```shell
python3 -m venv .venv
.venv/bin/pip install -U pip wheel future pymavlink mavproxy
```

and then I run everything from the virtualenv. This way I don't mess up the Homebrew-installed Python with random dependencies that are needed for ArduPilot only.

Reply:
I'm pretty sure it would help to put this in the instructions. Setting up a workable M1 Mac dev environment should be easy and without clear instructions it's definitely not. When you say "run everything from the virtualenv" - you mean mavproxy.py or waf or both? 

What I find now is that waf needs a different python environment to mavproxy, but sim_vehicle calls both waf and mavproxy so I can't get sim_vehicle->waf to use a different python from sim_vehicle->maxproxy.

For now in order to get sim_vehicle to work, I just run it 'headless', which seems like what @srmainwaring  and others are doing but I shouldn't have to do that.

Reply:
I have to admit I don't use `mavproxy` in the virtualenv because I'm using my own tools to talk to the simulator. `mavproxy` is problematic because it insists on bringing the entire wxPython distribution with it and I don't need that. wxPython currently provides pre-compiled wheels for Intel Macs only on PyPI, and only for Python 3.9, so until wxPython publishes newer wheels one would have to compile wxPython from source.

So, ultimately my goal was to get the build process running and to be able to run most of the non-GUI scripts from the `Tools` folder (e.g., the one that builds the bootloaders). For that, a virtualenv is fine, the only catch is that some of the scripts invoke other Python scripts or `waf` in a way that it does not respect the current Python interpreter setting (like [this one](https://github.com/ArduPilot/ardupilot/pull/20420) ) and just fall back to whatever Python is at the front of the system PATH. Activating the virtualenv from the shell with `source .venv/bin/activate` does the trick in 99% of the cases, for instance it ensures that `waf` picks up the correct (virtualenv-based) Python instead of the system Python.

_Edit_: I tried the wxPython snapshot builds from [here](https://wxpython.org/Phoenix/snapshot-builds/); it seems like they have `universal2` wheels for Python 3.9 there so there would be no need to compile wxPython from source, but it did not work for me so I just gave up on GUI stuff.